Cargo.toml

```diff
@@ -11,12 +11,21 @@ repository    = "https://github.com/chanced/jsonptr"
 version       = "0.4.7"
 
 [dependencies]
-serde      = { version = "1.0", default-features = false, features = ["alloc"] }
-serde_json = { version = "1.0", default-features = false, features = ["alloc"] }
+serde = { version = "1.0", default-features = false, features = ["alloc"] }
+serde_json = { version = "1.0", optional = true, default-features = false, features = [
+	"alloc",
+] }
+serde_yaml = { version = "0.9.34", optional = true }
+toml = { version = "0.8", optional = true }
 
 [features]
-default = ["std"]
+assign  = []
+default = ["std", "json", "resolve", "assign", "delete"]
```

Comment:
I'm only using `Resolve` in the json schema crate and I suspect that only needing `resolve` would be more common than needing `assign` and/or `replace`. 

Agreed that its a small crate though so this may not be worth the flag at all. Hmm.

Reply:
True... I think that'd be reasonable too. That said, I think it makes sense to split-off resolve if you do end up adding flags.

Reply:
I went ahead and made it granular with an `"ops"` flag that enables all. It may be a bit overkill but it should be fine.